src/extensions/surveys.tsx

```diff
@@ -219,3 +118,752 @@ export function generateSurveys(posthog: PostHog) {
+}) {
+    const textRef = Preact.createRef()
+    const [selectedChoices, setSelectedChoices] = useState<string | string[] | null>(
+        question.type === SurveyQuestionType.MultipleChoice ? [] : null
+    )
+    const [openChoiceSelected, setOpenChoiceSelected] = useState(false)
+    const [openEndedInput, setOpenEndedInput] = useState('')
+
+    const inputType = question.type === SurveyQuestionType.SingleChoice ? 'radio' : 'checkbox'
+    return (
+        <div
+            className="survey-box"
+            style={{ backgroundColor: appearance.backgroundColor || defaultBackgroundColor }}
+            ref={textRef}
+        >
+            <Cancel onClick={() => closeSurveyPopup()} />
+            <QuestionHeader
+                question={question.question}
+                description={question.description}
+                backgroundColor={appearance.backgroundColor}
+            />
+            <div className="multiple-choice-options">
+                {question.choices.map((choice: string, idx: number) => {
+                    let choiceClass = 'choice-option'
+                    const val = choice
+                    const option = choice
+                    if (!!question.hasOpenChoice && idx === question.choices.length - 1) {
+                        choiceClass += ' choice-option-open'
+                    }
+                    return (
+                        <div className={choiceClass}>
+                            <input
+                                type={inputType}
+                                id={`surveyQuestion${questionIndex}Choice${idx}`}
+                                name={`question${questionIndex}`}
+                                value={val}
+                                // disabled={!choice}
+                                onInput={() => {
+                                    if (question.hasOpenChoice && idx === question.choices.length - 1) {
+                                        setOpenChoiceSelected(!openChoiceSelected)
+                                    } else {
+                                        if (question.type === SurveyQuestionType.SingleChoice) {
+                                            setSelectedChoices(val)
+                                        } else if (
+                                            question.type === SurveyQuestionType.MultipleChoice &&
+                                            _isArray(selectedChoices)
+                                        ) {
+                                            if (selectedChoices.includes(val)) {
+                                                setSelectedChoices(selectedChoices.filter((choice) => choice !== val))
+                                            } else {
+                                                setSelectedChoices([...selectedChoices, val])
+                                            }
+                                        }
+                                    }
```

Comment:
My eyes 🙈 !!

Nested ifs are another source of bugs as well as being hard to parse. This is basically a switch statement so its way nicer to use returns to break the logic rather than having deeper and deeper nesting.

```suggestion
                                    if (question.hasOpenChoice && idx === question.choices.length - 1) {
                                        return setOpenChoiceSelected(!openChoiceSelected)
                                    }
                                    if (question.type === SurveyQuestionType.SingleChoice) {
                                        return setSelectedChoices(val)
                                    }

                                    if (
                                        question.type === SurveyQuestionType.MultipleChoice &&
                                        _isArray(selectedChoices)
                                    ) {
                                        if (selectedChoices.includes(val)) {
                                            return setSelectedChoices(
                                                selectedChoices.filter((choice) => choice !== val)
                                            )
                                        }
                                        return setSelectedChoices([...selectedChoices, val])
                                    }
```

Reply:
Furthermore - as soon as the logic is more than a one-liner or so you should pull it out to its own function above the JSX. Otherwise the markup becomes even more unreadable.